content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -0,0 +1,1682 @@
+
+<video controls loop><source  src="shadow_cascades.mp4" type="video/mp4"/></video>
+
+Notice how the nearby shadows are highly detailed whereas the shadows in the distance become less detailed as they get farther away (which doesn't matter as much because they are far away).
+
+While shadow cascades solve important problems, they also introduce new ones. How many cascades should you use? What is the minimum and maximum distance from the camera where shadows should appear? How much overlap should there be between cascades? These parameters must be dialed in to fit a given scene.
+
+## Environment Map Lighting
+
+<div class="release-feature-authors">authors: @JMS55</div>
+
+Environment maps are a popular and computationally cheap way to significantly improve the quality of a scene's lighting. It uses a cube map texture to provide 360 lighting "from all directions". This is especially apparent for reflective surfaces, but it applies to all lit materials.
+
+This is what the PBR material looks like without environment map lighting:
+
+![env map before](env_map_before.png)
+
+And this is what the PBR material looks like with environment map lighting:
+
+![env map after](env_map_after.png)
+
+For scenes that need constant lighting (especially outdoor scenes), environment maps are a great solution. And because environment maps are arbitrary images, artists have a lot of control over the character of the scene's lighting.
+
+## Depth and Normal Prepass
+
+<div class="release-feature-authors">authors: @icesentry, Rob Swain (@superdump), @robtfm, @JMS55</div>
+
+<video controls loop><source  src="force_field.mp4" type="video/mp4"/></video>
+<p class="release-feature-authors">This effect uses the depth from the prepass to find the intersection between the ground and the force field</p>
+
+Bevy now has the ability to run a depth and/or normal prepass. This means the depth and normal textures will be generated in a render pass that runs before the main pass and can therefore be used during the main pass. This enables various special effects like Screen Space Ambient Occlusion, Temporal Anti Aliasing and many more. These are currently being worked on and should be available in the next release of bevy.
+
+![Edge detection](edge_detection.png)
+<p class="release-feature-authors">In the image on the right, green lines are edges detected in the normal texture and blue lines are edges detected in the depth texture</p>
+
+![Edge detection prepass](edge_detection_prepass.png)
+<p class="release-feature-authors">The depth and normal textures generated by the prepass</p>
+
+Unfortunately, the prepass still has performance issues so it's currently disabled by default, but if you need to use it for a specific effect you can simply add the `DepthPrepass` component to your camera.
```

Comment:
Linking to an issue for the prepass performance issues would be nice for folks to track and help out with.

Reply:
There isn't one right now because it depends on many things that aren't all necessarily prepass related. Pretty much anything that makes rendering faster will also help the prepass.

Reply:
Something like this might help a little more

```suggestion
Using the prepass essentially means rendering everything twice. The prepass itself is much faster since it does a lot less work than the main pass. The result of the prepass can be used to reduce overdraw in the main pass, but if your scene didn't already suffer from overdraw then enabling the prepass will negatively affect performance.  There are many things that can be done to improve this and we will keep working towards this goal. Like with anything performance related, make sure to measure it for your use case and see if it helps or not.

The prepass is still very useful when working on special effects that require a depth or normal texture, so if you want to use it you can simply add the `DepthPrepass` or `NormalPrepass` components to your camera.
```

Reply:
I much prefer this. @cart .